compiler/src/utils/warnings.re

```diff
@@ -175,3 +175,25 @@ let check_fatal = () =>
     nerrors := 0;
     raise(Errors);
   };
+
+// track the warnings for the LSP
+let warnings_this_run: ref(list((loc, t))) = ref([]);
```

Comment:
We'll still want to reset this somewhere, since this will track all warnings across potentially unrelated programs that get compiled (you can think of the tests as a good example, since all of the tests get compiled individually without the compiler ever being "reset"). I think the place I suggested for this was `Typemod.type_implementation`, but with how you've changed it I think it might make more sense to do it in `compile.re` a similar way that we do `Env.clear_imports()`. It might also be useful to make a "reset" helper function that combines these.

Reply:
Oh, that's a good point.  I'd only tested individual programs and not see the warnings ever be carried across, but this gives me a good way to test now.  I'll generalise it into a reset helper too

Reply:
Just updated it with your suggestion @ospencer .  How does it look now?